apps/aecore/src/aec_chain_state.erl

```diff
@@ -131,12 +131,12 @@ median_timestamp(Header) ->
 
 -spec insert_block(aec_blocks:block() | map()) -> 'ok' | {'pof', aec_pof:pof()} | {'error', any()}.
 insert_block(#{ key_block := KeyBlock, micro_blocks := MicroBlocks, dir := forward }) ->
```

Comment:
This implies that all and only the places where this envelope `#{ key_block := _, micro_blocks := _, dir := forward | backward }` is in sync. I am finding this a little difficult to check (jumps in aec_sync module).

Would it make sense to check this envelope in aec_conductor:add_synced_block (for readability)?

Reply:
I quickly checked for `micro_blocks` and I found no points where this envelope is passed that is not sync-related so looks good.